server/channels/app/platform/web_conn.go

```diff
@@ -843,7 +883,17 @@ func (wc *WebConn) ShouldSendEvent(msg *model.WebSocketEvent) bool {
 	}
 
 	// Only report events to users who are in the channel for the event
-	if msg.GetBroadcast().ChannelId != "" {
+	if chID := msg.GetBroadcast().ChannelId; chID != "" {
+		// For typing events, we don't send them to users who don't have
+		// that channel or thread opened.
+		if wc.Platform.Config().FeatureFlags.WebSocketEventScope &&
+			utils.Contains([]model.WebsocketEventType{
+				model.WebsocketEventTyping,
+				model.WebsocketEventReactionAdded,
+			}, msg.EventType()) && wc.notInChannelAndThread(chID) {
```

Comment:
Can we simplify this to two equality checks and an or, instead of creating a slice + calling a function + looping + returning + garbage collecting? Considering this is a hot path?

Reply:
Ah yes, there was actually another event here, but I removed it :P So an OR makes sense now.